src/framework/ui/internal/mainwindow.cpp

```diff
@@ -87,3 +88,8 @@ QScreen* MainWindow::screen() const
 {
     return m_bridge ? m_bridge->screen() : nullptr;
 }
+
+mu::async::Notification MainWindow::isFullScreenChanged() const
+{
+    return m_isFullScreenChanged;
```

Comment:
This method just returns the notification of `MainWindow`, but `MainWindow` never sends this notification. As a result, nothing will happen. So, in `MainWindow::init`, you will need to subscribe to the notification from `m_bridge`, and when that notification is received, send the notification from `MainWindow`.

Bonus question: do you also understand why the following does ***not*** work?
```c++
mu::async::Notification MainWindow::isFullScreenChanged() const
 {
    return m_bridge ? m_bridge->isFullScreenChanged() : mu::async::Notification();
}
```